drivers/net/tun.c

```diff
@@ -986,6 +988,7 @@ static ssize_t tun_write(FAR struct file *filep, FAR const char *buffer,
         {
           net_lock();
           ret = netdev_iob_prepare(&priv->dev, false, 0);
+          priv->dev.d_buf = NULL;
```

Comment:
why this is needed?

Reply:
> why this is needed?

netdev_iob_prepare will set dev.d_buf = dev.d_iob->io_data, there will be problems the next devif_poll.
dev.d_buf should be NULL when offload case. this operation simply reset the dev.d_buf to previous value.